Feat/auth api

mjaveld opened the pull request:
Auth endpoint is done. Tests where used to implement the functionality. Resolves #36

mjaveld committed:
feat: WIP figuring out schema

mjaveld committed:
feat: easy script as reminder to set  testvars

mjaveld committed:
feat: Working and tested auth group controller

mjaveld committed:
feat: WIP auth user controller

mjaveld committed:
fix: add groupid to group struct

mjaveld committed:
feat: working user endpoint

mjaveld committed:
feat: Extra api key and otp seed endpoints working

mjaveld committed:
feat: working privilege search and get_item on auth endpoint

mjaveld committed:
Merge branch 'main' into feat/auth-api

browningluke commented on schema/auth.yml:
```diff
@@ -0,0 +1,274 @@
+      #     - name: id
+      #       optional: false
+      #       bodyParameter: false
+      #   result_type: SetItemPrivilegeWrapper
```
Are these commented out for a reason? Can we remove them if they're unused?

mjaveld commented on scripts/set_test_envvars.sh:
```diff
@@ -0,0 +1,3 @@
+export OPNSENSE_URI=https://192.168.1.96
+export OPNSENSE_API_KEY=P5xfc23KjJRsfsd2e9Tk93MRSyCpiJ6+XeXjerD6hUHDA4dxhWv0S3SaonopAwmGYrTthoJ+S4DhRcjp
+export OPNSENSE_API_SECRET=HtBjEp0DVmcpoTDECr8oNy62h8HiV0okoLgG2zPIOJkb7bogeITEhpyp2f1OE1t8dDjWmRg8VITWm1PK
```
Was an easy of use script to set the dev envvars against my homelab test instance. Its not required

mjaveld committed:
fix: removed dev envvar script

browningluke reviewed (changes requested):
Ah, this comment was left as 'Pending' on my end. Sorry, meant to leave it as a review.

This is my only issue, once this gets resolved this PR should be good to merge!

mjaveld committed:
feat: consistent naming for privilege structs

mjaveld committed:
fix: PrivilegeSetItem now working.

Still commented out because it needs feature from other branch to add customtype parameters

mjaveld committed:
fix: Fixed typo

mjaveld committed:
Merge branch 'main' into feat/auth-api

mjaveld committed:
feat: Privilege Set_item endpoint support and proper tests

browningluke reviewed (approved):
LGTM. Thanks again for this one too.
